### PR TITLE
EPMRPP-40028 || Test case id examples provided

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

**Review by pbortnik — APPROVED** [review]

File: example-cucumber/src/test/java/com/epam/reportportal/example/cucumber/Stepdefs.java [hunk 1]
```diff
@@ -9,17 +11,20 @@
 public class Stepdefs {
 	
 	private Belly belly = new Belly();
-	
+
+	@TestCaseId(isParameterized = true)
     @Given("^I have (\\d+) cukes in my belly$")
-    public void I_have_cukes_in_my_belly(int cukes) {
+    public void I_have_cukes_in_my_belly(@TestCaseIdKey(isInteger = true) int cukes) {
```
Comment (pbortnik) [inline example-cucumber/src/test/java/com/epam/reportportal/example/cucumber/Stepdefs.java]: update examples according to the new client changes

**Review by pbortnik — CHANGES_REQUESTED** [review]

File: example-cucumber/src/test/java/com/epam/reportportal/example/cucumber/Stepdefs.java (hunk 1 above)
Comment (Yumfriez) [reply example-cucumber/src/test/java/com/epam/reportportal/example/cucumber/Stepdefs.java]: updated

**Comment by DzmitryHumianiuk** [comment]
Expect to merge updates in `java-agents-*` first.

File: example-junit/src/test/java/com/epam/reportportal/example/junit/annotated/TestCaseIdParameterizedTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+public class TestCaseIdParameterizedTest implements ArtifactParams {
+	
+	@Rule
+	public final AtomIdentity identity = new AtomIdentity(this);
+
+	@TestCaseIdKey
+    private String input;
```
Comment (pbortnik) [inline example-junit/src/test/java/com/epam/reportportal/example/junit/annotated/TestCaseIdParameterizedTest.java]: formatting

File: example-testng-logback/src/main/java/com/epam/reportportal/example/testng/logback/logging/LoggingTest.java [hunk 3]
```diff
@@ -21,6 +24,7 @@ public void logCss() throws IOException {
         File file = File.createTempFile("rp-test", ".css");
         Resources.asByteSource(Resources.getResource("files/css.css")).copyTo(Files.asByteSink(file));
         LoggingUtils.log(file, "I'm logging CSS");
+//        throw new RuntimeException();
```
Comment (pbortnik) [inline example-testng-logback/src/main/java/com/epam/reportportal/example/testng/logback/logging/LoggingTest.java]: ?

**Review by pbortnik — CHANGES_REQUESTED** [review]

File: example-junit/src/test/java/com/epam/reportportal/example/junit/annotated/TestCaseIdParameterizedTest.java (hunk 2 above)
Comment (Yumfriez) [reply example-junit/src/test/java/com/epam/reportportal/example/junit/annotated/TestCaseIdParameterizedTest.java]: fixed

File: example-testng-logback/src/main/java/com/epam/reportportal/example/testng/logback/logging/LoggingTest.java (hunk 3 above)
Comment (Yumfriez) [reply example-testng-logback/src/main/java/com/epam/reportportal/example/testng/logback/logging/LoggingTest.java]: removed

File: example-cucumber/src/test/java/com/epam/reportportal/example/cucumber/Stepdefs.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+import com.epam.reportportal.annotations.TestCaseId;
+import com.epam.reportportal.annotations.TestCaseIdKey;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+
+import static org.junit.Assert.assertTrue;
+
+public class Stepdefs {
```
Comment (HardNorth) [inline example-cucumber/src/test/java/com/epam/reportportal/example/cucumber/Stepdefs.java:11]: please format the class

**Review by HardNorth — CHANGES_REQUESTED** [review]
Please also add examples for test case ids for ScenarioReporter for cucumber agent. I only see StepReporter approach.

File: example-cucumber/src/test/java/com/epam/reportportal/example/cucumber/Stepdefs.java (hunk 4 above)
Comment (Yumfriez) [reply example-cucumber/src/test/java/com/epam/reportportal/example/cucumber/Stepdefs.java:11]: fixed

**Review by HardNorth — APPROVED** [review]

**Pull request merged**